src/shared_utilities/interpolating_time_varying_input1d.jl

```diff
@@ -0,0 +1,138 @@
+import CUDA
+
+"""
+    NearestNeighbor
+
+Return the value corresponding to the point closest to the input time.
+"""
+struct NearestNeighbor <: AbstractInterpolationMethod end
+
+"""
+    LinearInterpolation
+
+Perform linear interpolation between the two neighboring points.
+"""
+struct LinearInterpolation <: AbstractInterpolationMethod end
+
+"""
+    InterpolatingTimeVaryingInput1D
+
+The constructor for InterpolatingTimeVaryingInput1D is not supposed to be used directly, unless you
+know what yo are doing. The constructor does not perform any check and does not take care of
+GPU compatibility. It is repsonsability of Integrator() to do so.
+
+"""
+struct InterpolatingTimeVaryingInput1D{
+    AA1 <: AbstractArray,
+    AA2 <: AbstractArray,
+    M <: AbstractInterpolationMethod,
+    CC <: ClimaComms.AbstractCommsContext,
+    R <: Tuple,
+} <: AbstractTimeVaryingInput
+    # AA1 and AA2 could be different because of different FTs
+
+    """Independent coordinate"""
+    times::AA1
+
+    """Variable"""
+    vals::AA2
+
+    """Interpolation method"""
+    method::M
+
+    """ClimaComms context"""
+    context::CC
+
+    """Range of times over which the interpolator is defined. range is always defined on the
+    CPU. Used by the in() function."""
+    range::R
+end
+
+# GPU compatibility
+function Adapt.adapt_structure(to, itp::InterpolatingTimeVaryingInput1D)
+    times = Adapt.adapt_structure(to, itp.times)
+    vals = Adapt.adapt_structure(to, itp.vals)
+    method = Adapt.adapt_structure(to, itp.method)
+    range = Adapt.adapt_structure(to, itp.range)
+    # On a GPU, we have a "ClimaCore.DeviceSideContext"
```

Comment:
This is called when you run a GPU kernel. It is needed to "adapt" a struct so that it can be moved to the GPU (e.g., possibly removing isbits objcets). In this case it is adapting the CuArrays.